File: typescript/src/batch_vehicle_signup.ts
import { createNodeClient } from "./client"
import { create, toJsonString } from "@bufbuild/protobuf";
import { EmptySchema } from "@bufbuild/protobuf/wkt";
import {
  BatchVehicleSignUpRequestSchema,
  BatchVehicleSignUpResponseSchema,
  Region
} from "@buf/nativeconnect_api.bufbuild_es/nativeconnect/api/v1/nativeconnect_pb";

const client = createNodeClient()

const request = create(BatchVehicleSignUpRequestSchema, {
  consentEmail: "consent_owner@email.com",
  consents: [
    {
      region: Region.COUNTRY_AUS,
      providerAuth: {
        vin: "LRW3F7FS9PC893577",
        additionalAuthInfo: { case: "empty", value: create(EmptySchema) }
      }
    },
    {
      region: Region.COUNTRY_AUS,
      providerAuth: {
        vin: "1C4RJHKG2P8863343",
        additionalAuthInfo: { case: "odometer", value: { odometer: 12345 } }
      }
    },
    {
      region: Region.COUNTRY_AUS,
      providerAuth: {
        vin: "JTMAAABJ004070717",
        additionalAuthInfo: {
          case: "usernamePassword",
          value: { username: "owner@example.com", password: "<<<PASSWORD>>>" }
        }
      }
    }
  ]
})

const response = await client.batchVehicleSignUp(request)

console.log(toJsonString(BatchVehicleSignUpResponseSchema, response, { prettySpaces: 2 }))

File: python/batch_vehicle_signup.py
from client import create_native_client
import nativeconnect.api.v1.nativeconnect_pb2 as native
from google.protobuf.empty_pb2 import Empty


def main():
    client = create_native_client()

    request = native.BatchVehicleSignUpRequest(
        consent_email="consent_owner@email.com",
        consents=[
            native.Consent(
                region=native.Region.COUNTRY_AUS,
                provider_auth=native.AuthRequest(
                    vin="LRW3F7FS9PC893577",
                    empty=Empty()
                )
            ),
            native.Consent(
                region=native.Region.COUNTRY_AUS,
                provider_auth=native.AuthRequest(
                    vin="1C4RJHKG2P8863343",
                    odometer=native.OdometerAuth(odometer=12345)
                )
            ),
            native.Consent(
                region=native.Region.COUNTRY_AUS,
                provider_auth=native.AuthRequest(
                    vin="JTMAAABJ004070717",
                    username_password=native.UsernamePasswordAuth(
                        username="owner@example.com",
                        password="<<<PASSWORD>>>"
                    )
                )
            ),
        ]
    )

    response = client.BatchVehicleSignUp(request)

    print(response)


if __name__ == "__main__":
    main()
